### ISSUE TITLE
Components.TryGet() is not clear that you will only get a component copy and not a modifiable ref

### ISSUE THREAD

**Issue by GamingRay** [outerminds/Entia.Unity#16]
```
if (HealthComponents.TryGet(entity, out var health))
{
    health.CurrentValue = 10;
}
```
This will compile and run without warning, but health component will never be modified. It is currently not really clear that this is the case and can cause unwanted time spent debugging.

**Comment by Magicolo** [comment]
Yes, the subtleties of using structs can be surprising sometimes. I will investigate a solution which will most likely take the form of a compiler warning.
